pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -9484,5 +9485,164 @@ spec:
+		pipeline: simplePipeline,
+		pipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: foo-pipeline-run
+  namespace: default
+spec:
+  pipelineRef:
+    name: foo-pipeline
+  taskRunSpecs:
+  - pipelineTaskName: 1st-task
+    computeResources:
+      limits:
+        cpu: "2"
+`),
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}, {
+		name:     "both with requests and limits",
+		pipeline: simplePipeline,
+		pipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: foo-pipeline-run
+  namespace: default
+spec:
+  pipelineRef:
+    name: foo-pipeline
+  taskRunSpecs:
+  - pipelineTaskName: 1st-task
+    computeResources:
+      requests:
+        cpu: "500m"
+      limits:
+        cpu: "2"
+`),
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}, {
+		name:     "both with cpu and memory",
+		pipeline: simplePipeline,
+		pipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: foo-pipeline-run
+  namespace: default
+spec:
+  pipelineRef:
+    name: foo-pipeline
+  taskRunSpecs:
+  - pipelineTaskName: 1st-task
+    computeResources:
+      requests:
+        cpu: "1"
+        memory: "1Gi"
+      limits:
+        cpu: "2"
+`),
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi")},
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{
+				PipelineRuns: []*v1beta1.PipelineRun{tc.pipelineRun},
+				Pipelines:    []*v1beta1.Pipeline{tc.pipeline},
+				Tasks:        []*v1beta1.Task{multiStepsTask},
+			}
+			prt := newPipelineRunTest(d, t)
+			defer prt.Cancel()
+
+			reconciledRun, clients := prt.reconcileRun("default", tc.pipelineRun.Name, []string{}, false)
+
+			if reconciledRun.Status.CompletionTime != nil {
+				t.Errorf("Expected a CompletionTime on valid PipelineRun, but got nil")
+			}
+
+			TaskRunList, err := clients.Pipeline.TektonV1beta1().TaskRuns("default").List(prt.TestAssets.Ctx, metav1.ListOptions{})
+			if err != nil {
+				t.Fatalf("Failure to list TaskRun's %s", err)
+			}
+
+			if err := verifyTaskLevelComputeResources(tc.expectedComputeResources, TaskRunList.Items); err != nil {
+				t.Errorf("TaskRun \"%s\" failed to verify task-level compute resource requirements, because: %v", tc.name, err)
+			}
+		})
+	}
+}
+
+// verifyTaskLevelComputeResources verifies that the created TaskRuns have the expected compute resources
+func verifyTaskLevelComputeResources(expectedComputeResources []corev1.ResourceRequirements, taskRuns []v1beta1.TaskRun) error {
```

Comment:
nit: since there is only one function that is using this helper, we could just move these lines within the test function itself (unless you are planning on adding new tests using the new helper)

Reply:
I see. let me update and also apply in the following work.